libraries/chain/resource_limits.cpp

```diff
@@ -441,4 +445,137 @@ account_resource_limit resource_limits_manager::get_account_net_limit_ex( const
    return arl;
 }
 
+void resource_limits_manager::update_proxied(int64_t now, symbol purpose_symbol, const account_name& account, int64_t frame_length, bool force) {
+    const auto& agents_idx = _db.get_mutable_index<stake_agent_index>().indices().get<stake_agent_object::by_key>();
+    const auto& grants_idx = _db.get_mutable_index<stake_grant_index>().indices().get<stake_grant_object::by_key>();
+    update_proxied_traversal(now, purpose_symbol, agents_idx, grants_idx, get_agent(purpose_symbol, agents_idx, account), 
+        frame_length, force);
+}
+
+void resource_limits_manager::recall_proxied(int64_t now, account_name grantor_name, account_name agent_name, 
+                    symbol_code token_code, symbol_code purpose_code, int16_t pct) {
+                        
+    EOS_ASSERT(1 <= pct && pct <= config::_100percent, transaction_exception, "pct must be between 0.01% and 100% (1-10000)");
+    const auto* param = _db.find<stake_param_object, by_id>(token_code.value);
+    EOS_ASSERT(param, transaction_exception, "no staking for token");
+    auto purpose_symbol = param->get_purpose_symbol(token_code, purpose_code);
+    
+    const auto& agents_idx = _db.get_mutable_index<stake_agent_index>().indices().get<stake_agent_object::by_key>();
+    const auto& grants_idx = _db.get_mutable_index<stake_grant_index>().indices().get<stake_grant_object::by_key>();
+
+    auto grantor_as_agent = get_agent(purpose_symbol, agents_idx, grantor_name);
+    
+    update_proxied_traversal(now, purpose_symbol, agents_idx, grants_idx, grantor_as_agent, param->frame_length, false);
+    
+    int64_t amount = 0;
+    auto grant_itr = grants_idx.lower_bound(grant_key(purpose_symbol, grantor_name));
+    while ((grant_itr != grants_idx.end()) &&
+           (grant_itr->purpose_id   == purpose_symbol.decimals()) &&
+           (grant_itr->token_code   == purpose_symbol.to_symbol_code()) &&
+           (grant_itr->grantor_name == grantor_name))
+    {
+        if (grant_itr->agent_name == agent_name) {
+            auto to_recall = get_prop(grant_itr->share, pct);
+            amount = recall_proxied_traversal(purpose_symbol, agents_idx, grants_idx, grant_itr->agent_name, to_recall, grant_itr->break_fee);
+            if (grant_itr->pct || grant_itr->share > to_recall) {
+                _db.modify(*grant_itr, [&](auto& g) { g.share -= to_recall; });
```

Comment:
yes, it can. in this case grantor has no share in agent balance (share == 0), but on receiving stake token in the future, pct will be delegate to the agent. pct is a user-defined parameter.

Reply:
ok